lib/Async/include/Async/async.h

```diff
@@ -46,8 +51,35 @@ struct async_promise_base : async_registry::AddToAsyncRegistry {
       void await_resume() noexcept {}
       async_promise_base* _promise;
     };
+    ExecContext::set(_callerExecContext);
     return awaitable{this};
   }
+  template<typename U>
+  auto await_transform(U&& other_awaitable) noexcept {
+    using inner_awaitable_type =
+        decltype(get_awaitable_object(std::forward<U>(other_awaitable)));
+    struct awaitable {
+      bool await_ready() { return inner_awaitable.await_ready(); }
+      auto await_suspend(std::coroutine_handle<> handle) {
+        promise->isSuspended = true;
+        ExecContext::set(promise->_callerExecContext);
+        return inner_awaitable.await_suspend(handle);
+      }
+      auto await_resume() {
+        if (promise->isSuspended) {
+          promise->_callerExecContext = ExecContext::currentAsShared();
+          promise->isSuspended = false;
+        }
+        ExecContext::set(promise->_myExecContext);
+        return inner_awaitable.await_resume();
+      }
+      async_promise_base<T>* promise;
+      inner_awaitable_type inner_awaitable;
+    };
+    _myExecContext = ExecContext::currentAsShared();
```

Comment:
`_myExecContext` is only used from within `awaitable`. There is no need for it to be a member of the `promise_type`. Instead we can just set the local member in `await_suspend` before suspending.

Reply:
I made `_myExecContext` a member of the awaitable.
We cannot just set this variable in `await_suspend`, because this does not need to be called before `await_resume`. E.g. for the `NoWait` test struct, only `await_resume` is called without any suspension. In `await_resume` we need to set the context to `myExecContext`, so this variable needs to be filled at that point. So the best option is to set `myExecContext` when the awaitable is created.